include/aws/http/connection.h

```diff
@@ -144,6 +144,38 @@ struct aws_http_proxy_options {
     struct aws_byte_cursor auth_password;
 };
 
+/**
+ * Options for changing the settings for an HTTP/2 connection.
+ */
+struct aws_http2_change_settings_options {
+    /**
+     * Optional
+     * The data of settings to change.
+     * Note: each setting has its boundary. If settings_array is not set, num_settings has to be 0 to send an empty
+     * SETTINGS frame.
+     */
+    struct aws_http2_setting *settings_array;
```

Comment:
bluhhhhh I know I told you to make the `aws_http2_change_settings_options` struct so we could re-use it between the function, and the connection_options. Now I think maybe we should put the function back to the way it was, and just add
`initial_settings_array`, `num_initial_settings`, and `on_initial_settings_ack` to the connection options.

I just have a sense I that the bindings will end up handling the callback differently for the change_settings() function vs the initial_settings. And if we're using the same struct, the bindings will end up making some class that represents that struct, and the binding APIs will end up weird ... or something

Also, I don't think we'll fire the callback if the connection fails to setup at all, so the documentation about "will always fire" won't be completely correct.